formatters/simplecov/legacy_formatter.go

```diff
@@ -0,0 +1,47 @@
+package simplecov
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/codeclimate/test-reporter/env"
+	"github.com/codeclimate/test-reporter/formatters"
+	"github.com/pkg/errors"
+)
+
+type resultSet struct {
+	Coverage map[string]formatters.Coverage `json:"coverage"`
+}
+
+func legacyFormat(r Formatter, rep formatters.Report) (formatters.Report, error) {
```

Comment:
Seems like we always return `rep`, can we avoid that and only return `error`?
And the caller deal with the return it needs to pass? Like `return rep, legacyFormat(r, rep)`?

Reply:
Might not be a `go` thing, though... let me know.

Reply:
I liked your suggestion so I tested it out, but for some unknown reason to me tests just started to fail without a clear explanation. Check https://github.com/codeclimate/test-reporter/pull/441/commits/dacd033784cd68e6b9b23275ccd6d7f2efc3233e , then I reverted the commit and tests passed again. So ...